### PR TITLE
Single page product List

### PR DESCRIPTION
**Pull request by m26unkwn**
1. Added the single Page Listing.

### CODE REVIEW

File: src/components/Navbar/index.js [hunk 1]
```diff
@@ -29,17 +29,19 @@ const Navbar = () => {
...
-            <img src={devLogo} alt='dev_logo' />
-          </button>
-        </NavLink>
-        <h1 className='logo-text'>DEVSTORE</h1>
+        <div className='flex flex-gap jc-center  ai-center'>
+          <NavLink to='/'>
+            <button className='btn btn-icon'>
+              <img src={devLogo} alt='dev_logo' />
+            </button>
+          </NavLink>
```
Comment (manojsarna) [inline src/components/Navbar/index.js:42]: If you arent using style classes Link would be suffice too

File: src/components/productList/Filter/index.js [hunk 2]
```diff
@@ -5,19 +5,29 @@ import { useData } from "../../../Context/stateManage/state-context";
 import { CheckboxBrand } from "./CheckboxBrand";
 
 const Filter = () => {
-  const { dispatch } = useData();
+  const {
+    dispatch,
+    state: { selectedFilters },
+  } = useData();
```
Comment (manojsarna) [inline src/components/productList/Filter/index.js:12]: Suggestion : using the same name dispatch can be confusing. Using specific naming conventions can help.

File: src/components/productList/utils/filters.js [hunk 3]
```diff
@@ -18,9 +18,9 @@ const category = [
   },
   {
     id: 3,
-    name: "T-shirts",
-    value: "tees",
-    label: "T-shirts",
+    name: "Mug",
+    value: "mug",
+    label: "Mug",
```
Comment (manojsarna) [inline src/components/productList/utils/filters.js:23]: Why are you using two different attributes with same value?

**Review by manojsarna — APPROVED** [review]
Overall the code looks good!!

UI Review : 

1. Search Component is still showing in Login / SignUp Page

image

2. Category Links Not Working Upon Clicking

File: src/components/productList/utils/filters.js (hunk 3 above)
Comment (m26unkwn) [reply src/components/productList/utils/filters.js:23]: Thanks, Manoj For pointing out that will resolve it.

File: src/components/productList/Filter/index.js (hunk 2 above)
Comment (m26unkwn) [reply src/components/productList/Filter/index.js:12]: I will change it.

File: src/components/Navbar/index.js (hunk 1 above)
Comment (m26unkwn) [reply src/components/Navbar/index.js:42]: yeah you can use any of the link or NavLink as the name suggest NavLink so I am using it for Navbar.

Comment (m26unkwn) [reply src/components/Navbar/index.js:42]: I have not added the category section yet so yeah it will not work now.

Comment (m26unkwn) [reply src/components/Navbar/index.js:42]: Search Component Will work if you are not login.

File: src/components/Navbar/index.js [hunk 4]
```diff
@@ -29,17 +29,19 @@ const Navbar = () => {
   return (
     <nav
       className={`nav-wrapper flex ${
-        pathArray.some((item) => currentPath.pathname === item)
+        pathArray.some((item) => currentPath.pathname === item) || token
           ? "jc-center"
           : "jc-between ai-center"
       } flex-gap`}>
```
Comment (saurabhchirde) [inline src/components/Navbar/index.js:35]: Try to make JSX clean by moving all the conditions outside the JSX.

File: src/components/Nuclie/ProductCard.js [hunk 5]
```diff
@@ -83,55 +83,60 @@ const ProductCard = (props) => {
...
-              Go To Cart
-            </button>
-          ) : (
+        <div className='badge pd-badge'>
+          <button
+            disabled={loading}
+            onClick={() =>
+              isProducInWishlist
+                ? removefromWishlistHandler(product._id)
+                : moveToWishlistHandler(product)
```
Comment (saurabhchirde) [inline src/components/Nuclie/ProductCard.js:102]: Here as well

File: src/components/Nuclie/ProductCard.js [hunk 6]
```diff
@@ -83,55 +83,60 @@ const ProductCard = (props) => {
...
+          <div className='pd-card-action pd-card-btn'>
             <button
               disabled={loading}
-              onClick={() => addToCartHandler(product)}
-              className={
-                loading ? "btn icon-text btn-disabled" : "btn icon-text"
-              }>
+              onClick={() =>
+                isProducInCart ? navigate("/cart") : addToCartHandler(product)
+              }
```
Comment (saurabhchirde) [inline src/components/Nuclie/ProductCard.js:131]: Write onclick function outside the JSX to make it cleaner.

File: src/components/SingleProduct/SingleProduct.js [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+          <div className='discount'>inclusive of all taxes</div>
+          <div className='card-divider'></div>
+          <div className='pd-card-action pd-card-btn flex flex-col'>
+            <button
+              disabled={actionDisable}
+              onClick={() =>
+                isProducInCart
+                  ? navigate("/cart")
+                  : addToCartHandler(SingleProduct.product)
+              }
```
Comment (saurabhchirde) [inline src/components/SingleProduct/SingleProduct.js:117]: Same here

File: src/components/SingleProduct/SingleProduct.js [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+              {isProducInCart ? "Go To Cart" : "Add To Cart"}
+            </button>
+
+            <button
+              disabled={actionDisable}
+              onClick={() =>
+                isProducInWishlist
+                  ? removefromWishlistHandler(product._id)
+                  : moveToWishlistHandler(product)
+              }
```
Comment (saurabhchirde) [inline src/components/SingleProduct/SingleProduct.js:131]: Here as well

**Review by saurabhchirde — APPROVED** [review]
Gave some suggestions, otherwise, your code is clean and easy to read. 
Good to merge after resolving those comments.

File: src/components/SingleProduct/SingleProduct.js (hunk 8 above)
Comment (m26unkwn) [reply src/components/SingleProduct/SingleProduct.js:131]: I will resolve it @saurabhchirde

**Pull request merged**